extensions/2.0/Khronos/KHR_techniques_webgl/README.md

```diff
@@ -0,0 +1,842 @@
+
+#### Values
+
+The `KHR_techniques_webgl.values` dictionary property defines the uniform values of a [`technique.uniform`](#uniforms) of the same key, and when specified, overrides the corresponding uniform `value`. The value supplied must conform to the `type` and `count` properties, if present, of the corresponding `Uniform` object, and must be present if no `value` is supplied in the referenced `Uniform` object.
+
+## Extension
+
+`KHR_techniques_webgl` is defined in the asset's top level `extensions` property with the following additional values.
+
+```json
+{
+    "extensions": {
+        "KHR_techniques_webgl": {
+            "programs": [
+                {
+                    "attributes": [
+                        "a_position"
+                    ],
+                    "fragmentShader": 0,
+                    "vertexShader": 1
+                }
+            ],
+            "shaders": [
+                {
+                    "type": 35632,
+                    "uri": "duck0FS.glsl"
+                },
+                {
+                    "type": 35633,
+                    "uri": "duck0VS.glsl"
+                }
+            ],
+            "techniques": [
+                {
+                    "program": 0,
+                    "attributes": {
+                        "a_position": {
+                            "type": 35665,
+                            "semantic": "POSITION"
+                        }
+                    },
+                    "uniforms": {
+                        "u_modelViewMatrix": {
+                            "type": 35676,
+                            "semantic": "MODELVIEW"
+                        }
+                    }
+                }
+            ]
+        }
+    }
+}
+```
+
+#### Programs
+
+GLSL shader programs are stored in the asset's `KHR_techniques_webgl.programs` property. This property contains one or more objects, one for each program.
+
+Each shader program includes an `attributes` property, which specifies the vertex attributes that will be passed to the shader, and the properties `fragmentShader` and `vertexShader`, which reference the index of the shader in the `KHR_techniques_webgl.shaders` property for the fragment and vertex shader GLSL source code, respectively.
+
+```json
+{
+    "programs": [
+        {
+            "attributes": [
+                "a_normal",
+                "a_position",
+                "a_texcoord0"
+            ],
+            "fragmentShader": 0,
+            "vertexShader": 1
+        }
+    ]
+}
+```
+
+#### Shaders
+
+One or more shader source files are listed in the asset's `KHR_techniques_webgl.shaders` property. Each shader specifies a `type` (vertex or fragment, defined as GL enum types) and either a `uri` to the file or a reference to a `bufferView`. Shader URIs may be URIs to external files or data URIs, allowing the shader content to be embedded as base64-encoded data in the asset. 
```

Comment:
The spec should mention the preferred MIME type for Data URIs with shader sources.